sample/redo/lib/main.dart

```diff
@@ -140,125 +142,122 @@ class _MyHomePageState extends State<MyHomePage> {
 
   Stream<Iterable<int>> _listIdStream() => _replicache.subscribe(
         (tx) async => (await tx.scan(prefix: '/list/')).map(
-          (item) => int.parse(item.key.substring('/list/'.length)),
+          (item) => item.value['id'],
         ),
       );
 
-  Stream<Iterable<Todo>> _todoStream() => _replicache.subscribe(
-        (tx) async => (await tx.scan(prefix: prefix)).map(
-          (item) => Todo.fromJson(item.value),
-        ),
-      );
+  Stream<Iterable<Todo>> _todoStream() => _replicache.subscribe(allTodosInTx);
 
   void _handleSync(bool syncing) {
     setState(() {
       _syncing = syncing;
     });
   }
 
-  Future<Todo> _read(ReadTransaction tx, int id) async {
+  static Future<Todo> _read(ReadTransaction tx, int id) async {
     final data = await tx.get(addPrefix(id));
     return data == null ? null : Todo.fromJson(data);
   }
 
-  Future<void> _write(dynamic tx, int id, Todo todo) {
-    return Future.error('Not implemented');
-    // return _replicache.put(addPrefix(key), todo.toJson());
+  static Future<void> _write(WriteTransaction tx, Todo todo) {
+    final key = addPrefix(todo.id);
+    return tx.put(key, todo);
+  }
+
+  static Future<bool> _del(WriteTransaction tx, int id) =>
+      tx.del(addPrefix(id));
+
+  void _selectListId(int listId) {
+    setState(() {
+      _selectedListId = listId;
+    });
   }
 
-  Future<void> _del(dynamic tx, int id) {
+  Future<void> _dropDatabase() async {
     throw UnimplementedError();
-    // return _replicache.del(addPrefix(key));
+    // Navigator.pop(context);
+    // var items = await _replicache.scan(prefix: prefix);
+    // for (final ScanItem item in items) {
+    //   await _replicache.del(item.key);
+    // }
+    // await _init();
   }
 
-  Future<void> _handleDone(int id, bool complete) async {
-    // TODO(arv): This should be mutate
-    _replicache.query((tx) async {
-      var todo = await _read(tx, id);
+  Mutator<void, Map<String, dynamic>> _todoCreateMutator;
+  Mutator<void, Map<String, dynamic>> _todoDoneMutator;
+  Mutator<void, Map<String, dynamic>> _todoRemoveMutator;
+  Mutator<void, Map<String, dynamic>> _todoReorderMutator;
+
+  _registerMutations() {
```

Comment:
If you register these globally, it will be a smaller code change too because you can just do:

```
Mutator<void, Map<String, dynamic>> = _replicache.register('todo-create', ...
```
```

Reply:
Not sure what you mean?

Reply:
It only made sense with the suggestion to register the mutators globally at the top of main.dart.